### ISSUE TITLE
eqeqeq question

### ISSUE THREAD

**Issue by simov** [eslint/eslint#6308]
Just a question:

Is it a good idea to have even smarter rule to exclude things like:

```js
somevar == 42
```

I mean what possibly can go wrong with that? In cases when `somevar` comes from a querystring parsing that definitely would be parsed as string, yet most probably that would be what you want to check for.

**Comment by platinumazure** [comment]
Hmm... by the same token, couldn't you compare against `"42"` or use `parseInt()`, if you know a string is incoming in that case?

**Comment by simov** [comment]
Yep these are all possible conditions:

```js
somevar === 42 || somevar === '42'
Math.round(somevar) === 42
parseInt(somevar, 10) === 42
/^42$/.test(somevar)
```

In my case though `somevar` can be either `1`, `'1'`, `2`, `'2'` or `'somestring'`.

Actually just by looking at this [table](http://dorey.github.io/JavaScript-Equality-Table/) it turns out that the `==` comparison with integer can be broken by some really weird cases like:

```js
true==1 // obviously
[1]==1 // :trollface:
```

**Comment by platinumazure** [comment]
Well, don't forget you can always suppress a rule inline as well:

```js
if (queryStringParam == 42) { // eslint-disable-line eqeqeq
    // Do something
}
```

**Comment by simov** [comment]
Thanks for the tip! I guess that won't be a good feature after all. Especially in cases when you compare with 0 or 1.
